src/googleError.ts

```diff
@@ -55,13 +55,45 @@ export class GoogleError extends Error {
     }
     return err;
   }
+
+  // Parse http JSON error and promote google.rpc.ErrorInfo if exist.
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  static parseHttpError(json: any): GoogleError {
+    const error = Object.assign(
+      new GoogleError(json['error']['message']),
+      json.error
+    );
+    // Keep consistency with gRPC statusDetails fields. gRPC details has been occupied before.
+    // Rename "detials" to "statusDetails".
+    error.statusDetails = json['error']['details'];
+    delete error.details;
+    // Promote the ErrorInfo fields as first-class citizen in error.
```

Comment:
The term "first-class citizen" should not be used as non-inclusive, please consider changing to "top level".

Reply:
sg. Thank you for sharing.